### PR TITLE
feat(daemon & cli) support child tasks

### PR DESCRIPTION
**Pull request by lwander**
This lets us deploy all spinnaker components in parallel, and then surface it nicely in the UI: 

actuate1

actuate2

actuate3

### CODE REVIEW

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/ui/v1/AnsiSnippet.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 package com.netflix.spinnaker.halyard.cli.ui.v1;
 
 import lombok.Data;
+import org.springframework.security.access.method.P;
```
Comment (duftler) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/ui/v1/AnsiSnippet.java]: Accidentally added?

File: halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/tasks/v1/DaemonTask.java [hunk 2]
```diff
@@ -57,4 +64,31 @@ public boolean isTerminal() {
...
+    DaemonTask child = TaskRepository.submitTask(childRunner, name);
+    children.add(child);
+    return child;
+  }
+
+  public <P> DaemonResponse<P> reapChild(String childUuid) {
+    DaemonTask childTask = children.stream()
+        .filter(c -> c.getUuid().equals(childUuid))
+        .findFirst()
+        .orElseThrow(() -> new IllegalArgumentException("Child " + childUuid + " not belong to this task (" + uuid + ")"));
```
Comment (duftler) [inline halyard-core/src/main/java/com/netflix/spinnaker/halyard/core/tasks/v1/DaemonTask.java]: Missing a word in this exception message.

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/DistributedService.java [hunk 3]
```diff
@@ -98,8 +98,8 @@ default Integer getLatestEnabledServiceVersion(AccountDeploymentDetails<A> detai
     // 2. Something is wrong, so you rollback.
     // 3. Fixing the bad server group requires redeploying.
     //
-    // Since you can't fix the newest destroyd server group in place, and you (at least I cant imagine why)
-    // won't want to reenable that server group, there is no point it keeping it around. There's an argument
+    // Since you can't fix the newest destroyd server group in place, and you won't (at least I cant imagine why)
```
Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/DistributedService.java]: s/destroyd/destroyed

Comment (duftler) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/DistributedService.java]: s/cant/can't

File: halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/AccountController.java [hunk 4]
```diff
@@ -115,7 +115,7 @@
     builder.setFieldOptionsResponse(() -> accountService.getAccountOptions(deploymentName, providerName, accountName, fieldName));
     builder.setSeverity(severity);
 
-    return TaskRepository.submitTask(builder::build);
+    return DaemonTaskHandler.submitTask(builder::build, "Get " + fieldName + " options");
```
Comment (duftler) [inline halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/AccountController.java]: PUT == 'Get ...'?

File: halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/AccountController.java [hunk 5]
```diff
@@ -99,7 +99,7 @@
...
```
Comment (duftler) [inline halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/AccountController.java]: POST == 'Get ...'?

Comment (lwander) [inline halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/AccountController.java]: Yeah, GET requests can't have bodies, and both of these options endpoints you commented on need to supply state to stage for checking what options are available.

**Pull request merged**